test/DGmethods/cfl.jl

```diff
@@ -0,0 +1,149 @@
+
+using CLIMA.VariableTemplates: Vars
+using StaticArrays
+
+function initialcondition!(state, aux, coords, t)
+  FT = eltype(state)
+
+  p∞::FT = 10 ^ 5
+  T∞::FT = 300
+  translation_speed::FT = 150
+  translation_angle::FT = pi / 4
+  α = translation_angle
+  u∞ = SVector(translation_speed * cos(α), translation_speed * sin(α), 0)
+
+  u = u∞
+  T = T∞
+  # adiabatic/isentropic relation
+  p = p∞ * (T / T∞) ^ (FT(1) / kappa_d)
+  ρ = air_density(T, p)
+
+  state.ρ = ρ
+  state.ρu = ρ * u
+  e_kin = u' * u / 2
+  state.ρe = ρ * total_energy(e_kin, FT(0), T)
+
+  nothing
+end
+
+
+let
+  # boiler plate MPI stuff
+  CLIMA.init()
+  mpicomm = MPI.COMM_WORLD
+  ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+  loglevel = ll == "DEBUG" ? Logging.Debug :
+  ll == "WARN"  ? Logging.Warn  :
+  ll == "ERROR" ? Logging.Error : Logging.Info
+  logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+  global_logger(ConsoleLogger(logger_stream, loglevel))
+
+  # Mesh generation parameters
+  N = 4
+  Nq = N+1
+  Neh = 10
+  Nev = 4
+
+  @testset "$(@__FILE__) DGModel matrix" begin
+    for FT in (Float64, Float32)
+      for dim = (2, 3)
+        if dim == 2
+          brickrange = (range(FT(0); length=Neh+1, stop=1),
+                        range(FT(1); length=Nev+1, stop=2))
+        elseif dim == 3
+          brickrange = (range(FT(0); length=Neh+1, stop=1),
+                        range(FT(0); length=Neh+1, stop=1),
+                        range(FT(1); length=Nev+1, stop=2))
+        end
+
+        topl = StackedBrickTopology(mpicomm, brickrange)
+
+        function warpfun(ξ1, ξ2, ξ3)
+          FT = eltype(ξ1)
+
+          ξ1 ≥ FT(1//2) && (ξ1 = FT(1//2) + 2(ξ1 - FT(1//2)))
+          if dim == 2
+            ξ2 ≥ FT(3//2) && (ξ2 = FT(3//2) + 2(ξ2 - FT(3//2)))
+          elseif dim == 3
+            ξ2 ≥ FT(1//2) && (ξ2 = FT(1//2) + 2(ξ2 - FT(1//2)))
+            ξ3 ≥ FT(3//2) && (ξ3 = FT(3//2) + 2(ξ3 - FT(3//2)))
+          end
+          (ξ1, ξ2, ξ3)
+        end
+
+        grid = DiscontinuousSpectralElementGrid(topl,
+                                                FloatType = FT,
+                                                DeviceArray = ArrayType,
+                                                polynomialorder = N,
+                                                meshwarp = warpfun)
+
+        model = AtmosModel(NoOrientation(),
+                           NoReferenceState(),
+                           ConstantViscosityWithDivergence(FT(0)),
+                           DryModel(),
+                           NoRadiation(),
+                           NoSubsidence{FT}(),
+                           Gravity(),
+                           PeriodicBC(),
+                           initialcondition!)
+
+        dg = DGModel(model, grid, Rusanov(), CentralNumericalFluxDiffusive(),
+                     CentralGradPenalty())
+
+        Δt = FT(1//2)
+
+        function wavespeed_dt(m::AtmosModel, state::Vars, aux::Vars,
```

Comment:
should we make this a function in `balancelaw.jl` and move the method into `AtmosModel.jl`?

Reply:
I don't think so since.  As I understand it, @fxgiraldo wants to compute the effective CFL for various wave speed functions, not just one authoritative one.

Reply:
Ah, ok. We could make this the default though.

Reply:
Sure, I am open to that.

Reply:
What will primarily need in the end is the CFL number based on the advective velocity. The speed of sound only varies by ~20% in the atmosphere, so the acoustic Courant number can be computed at initialization time and will not vary much. 

It would help to add a diffusive CFL number too (can easily be accomplished with this set of functions by making the input more general than wavespeed * dt). This will be limiting in many cases.